src/Network/Network/VirtualNetwork/Peering/SyncAzureVirtualNetworkPeeringCommand.cs

```diff
@@ -0,0 +1,119 @@
+            ParameterSetName = FieldsParameterSetName)]
+        [ResourceNameCompleter("Microsoft.Network/virtualNetworks/virtualNetworkPeerings", "ResourceGroupName", "VirtualNetworkName")]
+        [ValidateNotNullOrEmpty]
+        [SupportsWildcards]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = "The virtual network peering",
+            ParameterSetName = ObjectParameterSetName)]
+        public PSVirtualNetworkPeering VirtualNetworkPeering { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            PSVirtualNetworkPeering existingPeering = null;
+
+            if (this.ParameterSetName == FieldsParameterSetName)
+            {
+                existingPeering = this.GetVirtualNetworkPeering(this.ResourceGroupName, this.VirtualNetworkName, this.Name);
+            }
+            else
+            {
+                // Piping scenario
+                existingPeering = this.VirtualNetworkPeering;
+            }
+
+            var updatedPeering = UpdateVirtualNetworkPeering(existingPeering);
+            WriteObject(updatedPeering);
+        }
+
+        private PSVirtualNetworkPeering UpdateVirtualNetworkPeering(PSVirtualNetworkPeering existingPeering)
+        {
+            if (!this.IsVirtualNetworkPeeringPresent(existingPeering.ResourceGroupName, existingPeering.VirtualNetworkName, existingPeering.Name))
+            {
+                throw new AzPSArgumentException(Microsoft.Azure.Commands.Network.Properties.Resources.ResourceNotFound, nameof(existingPeering.Name));
+            }
+
+            // Map to the sdk object
+            var vnetPeeringModel = NetworkResourceManagerProfile.Mapper.Map<MNM.VirtualNetworkPeering>(existingPeering);
+
+            //Get the remote VNet Id and then get the token for the resource Id if tis in a different tenant
+            var remoteVirtualNetworkId = existingPeering.RemoteVirtualNetwork.Id;
+            Dictionary<string, List<string>> auxAuthHeader = null;
+            if (remoteVirtualNetworkId != null)
+            {
+                List<string> resourceIds = new List<string>();
+                resourceIds.Add(remoteVirtualNetworkId);
+                var auxHeaderDictionary = GetAuxilaryAuthHeaderFromResourceIds(resourceIds);
+                if (auxHeaderDictionary != null && auxHeaderDictionary.Count > 0)
+                {
+                    auxAuthHeader = new Dictionary<string, List<string>>(auxHeaderDictionary);
+                }
+            }
+
+            // Create/Update call with 'sync' param set to true
+            this.VirtualNetworkPeeringClient.CreateOrUpdateWithHttpMessagesAsync(existingPeering.ResourceGroupName, existingPeering.VirtualNetworkName, existingPeering.Name, vnetPeeringModel, true.ToString(), auxAuthHeader).GetAwaiter().GetResult();
```

Comment:
this class should have "supportshouldprocess = true" because this PUT request. Please also remove breaking changes message added.

Reply:
I have removed the breaking changes message.

Why is supportsshouldprocess = true required?